src/components/AttachmentCarousel.js

```diff
@@ -0,0 +1,204 @@
+            file,
+            isBackDisabled: page === 0,
+            isForwardDisabled: page === attachments.length - 1,
+        });
+    }
+
+    /**
+     * increments or decrements the index to get another selected item
+     * @param {Number} deltaSlide
+    */
+    cycleThroughAttachments(deltaSlide) {
+        if ((deltaSlide < 0 && this.state.isBackDisabled) || (deltaSlide > 0 && this.state.isForwardDisabled)) {
+            return;
+        }
+        this.setState(({attachments, page}) => {
+            const nextIndex = page + deltaSlide;
+            const {sourceURL, file} = this.getAttachment(attachments[nextIndex]);
+            this.props.onArrowPress({sourceURL, file});
+            return {
+                page: nextIndex,
+                sourceURL,
+                file,
+                isBackDisabled: nextIndex === 0,
+                isForwardDisabled: nextIndex === attachments.length - 1,
+            };
+        });
+    }
+
+    render() {
+        if (!this.state.sourceURL) {
+            return null;
+        }
+
+        return (
+            <View
+                style={[styles.attachmentModalArrowsContainer]}
+                onMouseEnter={() => this.onShowArrow(true)}
+                onMouseLeave={() => this.onShowArrow(false)}
+            >
+                {this.state.showArrows && (
+                    <>
+                        <Button
+                            medium
+                            style={[styles.leftAttachmentArrow]}
+                            icon={Expensicons.BackArrow}
+                            iconFill={themeColors.text}
+                            iconStyles={[styles.mr0, {PointerEvent: 'auto'}]}
+                            onPress={() => this.cycleThroughAttachments(-1)}
+                            isDisabled={this.state.isBackDisabled}
+                        />
+                        <Button
+                            medium
+                            style={[styles.rightAttachmentArrow]}
+                            icon={Expensicons.ArrowRight}
+                            iconFill={themeColors.text}
+                            iconStyles={[styles.mr0, {PointerEvent: 'auto'}]}
+                            onPress={() => this.cycleThroughAttachments(1)}
+                            isDisabled={this.state.isForwardDisabled}
+                        />
+                    </>
+                )}
+
+                <Carousel
+                    styles={[styles.attachmentModalArrowsContainer]}
+                    canSwipeLeft={!this.state.isBackDisabled}
+                    canSwipeRight={!this.state.isForwardDisabled}
+                    onPress={() => this.canUseTouchScreen && this.onShowArrow(!this.state.showArrows)}
+                    onSwipe={this.cycleThroughAttachments}
+                >
+                    <AttachmentView onPDFPress={() => this.onShowArrow(!this.state.showArrows)} sourceURL={this.state.sourceURL} file={this.state.file} />
```

Comment:
We could listen to tap either at Attachment view or Carousel why keep things at multiple places?

Reply:
This is because I need to drill down the listener callback to the pdf. It seems the pdf view subverts the carousel listeners.

Reply:
> This is because I need to drill down the listener callback to the pdf.

Can you rephrase that?

Reply:
in order for the pdf to receive onPress events I had to send this through the AttachmentView as well. I was unable to find any better way than this.

Reply:
I tried that by wrapping ImageView in a Pressable
```
            <Pressable onPress={props.onPress}>
                <ImageView url={props.sourceURL} />
            </Pressable>
```

but the events from that onPress won't bubble up to the CarouselActions and I'm unable to get swiping to work.